### PR TITLE
kubeadm: don't re-add an etcd member if it already exists on "join"

### PR DESCRIPTION
**Pull request by neolit123**
**What this PR does / why we need it**:

If an etcd member with the same address already exists, don't re-add it.
Instead, use the existing member list for creating the "initial cluster"
that is written for this etcd server instance static Pod.

**Which issue(s) this PR fixes**:

xref https://github.com/kubernetes/kubeadm/issues/2095

**Special notes for your reviewer**:
NONE

**Does this PR introduce a user-facing change?**:

```release-note
kubeadm: during "join", don't re-add an etcd member if it already exists in the cluster.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/phases/etcd/local.go [hunk 1]
```diff
@@ -149,16 +149,29 @@ func CreateStackedEtcdStaticPodManifestFile(client clientset.Interface, manifest
...
+	initialCluster, err := etcdClient.ListMembers()
 	if err != nil {
 		return err
 	}
-	fmt.Println("[etcd] Announced new etcd member joining to the existing etcd cluster")
-	klog.V(1).Infof("Updated etcd member list: %v", initialCluster)
+
+	// only add the new member if it doesn't already exists
+	klog.V(1).Infof("[etcd] Checking if the etcd member already exists: %s", etcdPeerAddress)
+	id, _ := etcdClient.GetMemberID(etcdPeerAddress)
```
Comment (rosti) [inline cmd/kubeadm/app/phases/etcd/local.go:162]: Both `ListMembers()` and `GetMemberID()` call `listMembers()`. There's no need to do that twice (especially if we are trying to solve a problem that presents itself due to loss of connectivity).
Instead, you can avoid calling `GetMemberID` by just checking if `etcdPeerAddress` matches an entry in `initialCluster`.

**Review by rosti — CHANGES_REQUESTED** [review]
Thanks @neolit123 !

File: cmd/kubeadm/app/phases/etcd/local.go (hunk 1 above)
Comment (neolit123) [reply cmd/kubeadm/app/phases/etcd/local.go:162]: that is true. i will update this later today.

Comment (neolit123) [reply cmd/kubeadm/app/phases/etcd/local.go]: updated.

**Review by rosti — APPROVED** [review]
Thanks @neolit123 !
/lgtm
/hold cancel
/retest

**Pull request merged**